landoapi/hgexports.py

```diff
@@ -163,6 +163,129 @@ def write_commit_description(self, f: io.BytesIO):
         """Writes the commit description to the specified file object."""
         f.write(self.commit_description())
 
+    def get_diff(self) -> bytes:
+        """Return the diff for this patch."""
+        try:
+            diff = []
+            line_no = 0
+            for line in self.patch:
+                line_no += 1
+
+                if self.diff_start_line:
+                    if line_no == self.diff_start_line:
+                        diff.append(line)
+                        break
+                else:
+                    if self._is_diff_line(line):
+                        diff.append(line)
+                        break
+
+            while 1:
+                buf = self.patch.read(16 * 1024)
```

Comment:
This is just an issue with the diff display, this code is the already-existing code in `PatchHelper`. I can address this in a separate patch.

Reply:
Noted, RE: `16 * 1024`. However, the other comment still applies (about the large try/except and the while loop). I think since this code is being modified, it makes sense to clean it up while you're here. Parts of it are also copied to the base class definition of `write`, so at the very least that definition could benefit from a refactoring.